### ISSUE TITLE
Unexpected Java version in openmicrocopy/omero-server Docker image

### ISSUE THREAD

**Issue by sbesson** [ome/omero-server-docker#45]
Discovered as part of an investigation on OMERO 5.6 with JDK 11

The installation playbook currently install JDK 11 by setting the `java_versions` to `11` only - see https://github.com/ome/omero-server-docker/blob/5.6.2/playbook.yml#L5

However,  when consuming the image in the context of https://github.com/ome/docker-example-omero/blob/master/docker-compose.yml for instance, it looks like the default Java 8 version is still used:

```
[sbesson@idr2-slot3 docker-example-omero]$ docker exec -it docker-example-omero_omeroserver_1 bash
bash-4.2$ java -version
openjdk version "1.8.0_242"
OpenJDK Runtime Environment (build 1.8.0_242-b08)
OpenJDK 64-Bit Server VM (build 25.242-b08, mixed mode)
bash-4.2$ /opt/omero/server/OMERO.server/bin/omero admin diagnostics

================================================================================
OMERO Diagnostics (admin) 5.6.2
================================================================================
        
Commands:   java -version                  1.8.0     (/usr/bin/java)
Commands:   python -V                      3.6.8     (/opt/omero/server/venv3/bin/python -- 2 others)
...
```

Trying to override `JAVA_HOME` before starting the Docker container did not solve the issue

```diff
diff --git a/docker-compose.yml b/docker-compose.yml
index 60fd75c..fcf79da 100644
--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -21,11 +21,12 @@ services:
       CONFIG_omero_db_pass: omero
       CONFIG_omero_db_name: omero
       ROOTPASS: omero
+      JAVA_HOME: /usr/lib/jvm/java-11-openjdk-11.0.6.10-1.el7_7.x86_64/bin/java
```

It looks like Java 8 is getting pulled from another dependency as it is not part of the base `centos:7` image. Ideally it should be tracked down and removed/made configurable to the version set by the `ome.java` role.

At the API level, I can see two possible behaviors for the base `omero-server` image:

- either supporting multiple concurrent of Java. In that case, a secondary issues is to be abel to dynamically select which one should be used at runtime
- or support one Java version, and generally one deployment environment, for the base `omero-server` image and impose a rebuild with different parameters in `playbook.yml` when another version is desired.

**Comment by sbesson** [comment]
After increasing the verbosity using `ansible-playbook -v playbook.yml`, I confirmed a prior suggestion from @manics that the `ice-all-runtime` meta-package is the culprit for pulling in the Java8 dependencies when installing the server via Ansible:

```
yum[sbesson@ome-demoserver ~]$ yum deplist ice-all-runtime
Loaded plugins: langpacks, product-id, rhnplugin, search-disabled-repos,
              : subscription-manager
*Note* Spacewalk repositories are not listed below. You must run this command as root to access Spacewalk repositories.
Repo rhel-7-server-rpms forced skip_if_unavailable=True due to: /etc/pki/entitlement/5900882248084321346-key.pem
Repo rhel-7-server-optional-rpms forced skip_if_unavailable=True due to: /etc/pki/entitlement/5900882248084321346-key.pem
package: ice-all-runtime.i686 3.6.5-1.el7
  dependency: icebox(x86-32) = 3.6.5-1.el7
   provider: icebox.i686 3.6.5-1.el7
  dependency: libicestorm3.6(x86-32) = 3.6.5-1.el7
   provider: libicestorm3.6.i686 3.6.5-1.el7
package: ice-all-runtime.x86_64 3.6.5-1.el7
  dependency: glacier2(x86-64) = 3.6.5-1.el7
   provider: glacier2.x86_64 3.6.5-1.el7
  dependency: ice-utils-java = 3.6.5-1.el7
   provider: ice-utils-java.noarch 3.6.5-1.el7
  dependency: icebox(x86-64) = 3.6.5-1.el7
   provider: icebox.x86_64 3.6.5-1.el7
  dependency: icegrid(x86-64) = 3.6.5-1.el7
   provider: icegrid.x86_64 3.6.5-1.el7
  dependency: icepatch2(x86-64) = 3.6.5-1.el7
   provider: icepatch2.x86_64 3.6.5-1.el7
  dependency: libfreeze3.6-c++(x86-64) = 3.6.5-1.el7
   provider: libfreeze3.6-c++.x86_64 3.6.5-1.el7
  dependency: libice3.6-c++(x86-64) = 3.6.5-1.el7
   provider: libice3.6-c++.x86_64 3.6.5-1.el7
  dependency: libicestorm3.6(x86-64) = 3.6.5-1.el7
   provider: libicestorm3.6.x86_64 3.6.5-1.el7
  dependency: php-ice(x86-64) = 3.6.5-1.el7
   provider: php-ice.x86_64 3.6.5-1.el7
[sbesson@ome-demoserver ~]$ yum deplist ice-utils-java 
Loaded plugins: langpacks, product-id, rhnplugin, search-disabled-repos, subscription-manager
*Note* Spacewalk repositories are not listed below. You must run this command as root to access Spacewalk repositories.
Repo rhel-7-server-rpms forced skip_if_unavailable=True due to: /etc/pki/entitlement/5900882248084321346-key.pem
Repo rhel-7-server-optional-rpms forced skip_if_unavailable=True due to: /etc/pki/entitlement/5900882248084321346-key.pem
package: ice-utils-java.noarch 3.6.5-1.el7
  dependency: /bin/sh
   provider: bash.x86_64 4.2.46-30.el7
  dependency: java
   provider: java-1.8.0-openjdk.x86_64 1:1.8.0.191.b12-0.el7_5
   provider: java-1.8.0-openjdk.i686 1:1.8.0.191.b12-0.el7_5
   provider: java-1.7.0-openjdk.x86_64 1:1.7.0.191-2.6.15.4.el7_5
```

In total, this isssue is not specific to this Docker repository but affecting the entire Ansible stack could be migrated down to https://github.com/ome/ansible-role-ice. As per the above and https://doc.zeroc.com/rel/ice-releases/ice-3-6/ice-3-6-4-release-notes/using-the-linux-binary-distributions-for-ice-3-6-4#UsingtheLinuxBinaryDistributionsforIce3.6.4-rpmRPMPackages, one consideration is whether the meta-package should be replaced by a more fine-grained list of packages excluding `ice-utils-java` by default assuming the latter should be managed by `ome.java`.

**Comment by jburel** [comment]
I assume we will need to review the omero-install instructions too since we use ``yum -y install ice-all-runtime`` for C7

**Comment by sbesson** [comment]
:+1: I didn't realise it was also in `omero-install` and the walkthroughs. Certainly worth reviewing/updating all instances of https://github.com/search?q=org%3Aome+ice-all-runtime&type=Code wherever we want JDK 11 deployments
